readline/readline.go

```diff
@@ -116,19 +114,9 @@ func (i *Instance) Readline() (string, error) {
 
 			switch r {
 			case KeyUp:
-				if i.History.Pos > 0 {
-					if i.History.Pos == i.History.Size() {
-						currentLineBuf = []rune(buf.String())
-					}
-					buf.Replace([]rune(i.History.Prev()))
-				}
```

Comment:
This and the block below are unnecessary since `History.Prev()` and `History.Next()` should return the correct line content; the logic is duplicated between here and `readline/history.go` and it seems more appropriate for it to live there. The only benefit of having this conditional is omitting the call to `Replace()` which doesn't seem worth it for the added complexity